flood_api/utils/validation_helpers.py

```diff
@@ -43,7 +83,43 @@ def validate_dates(start_date: date, end_date: date) -> None:
     Returns:
     None
     """
-    date_range_valid = start_date <= end_date
+    date_range_is_valid = start_date <= end_date
 
-    if not date_range_valid:
-        raise HTTPException(status_code=400, detail="Invalid date range")
+    if not date_range_is_valid:
+        raise HTTPException(
+            status_code=INVALID_STATUS_CODE, detail="Invalid date range"
+        )
+
+
+def validate_generic_inputs(*args: tuple, possible_inputs_str: str) -> None:
```

Comment:
I'm not sure if we need a function like this. It seems quite complicated, but all it does is check if only coordinates or only bbox is present right, and then some error handling? I think it would be much simpler to do this in the if-elif that you already have in the router.

If you want to make something fancy I would consider making a FastAPI-dependency instead that uses both the coordinates and bbox dependencies. Since we are using Python 3.11, the return type of the dependency can then be a union type which we inside the router can use [match-statements](https://peps.python.org/pep-0622/) to unwrap:

```python
match query_shape:
    case lon, lat:
        ...
    case min_lat, max_lat, min_lon, max_lon:
        ...

Reply:
Created a new dependency for both types 👍